scalalib/src/mill/scalalib/JavaModule.scala

```diff
@@ -1131,7 +1131,15 @@ trait JavaModule
   )
 
   @internal
-  def bspJvmBuildTarget: Task[JvmBuildTarget] = Task.Anon {
+  @deprecated("Use bspJvmBuildTargetTask instead", "0.12.3")
+  def bspJvmBuildTarget: JvmBuildTarget =
+    JvmBuildTarget(
+      javaHome = Option(System.getProperty("java.home")).map(p => BspUri(os.Path(p))),
+      javaVersion = Option(System.getProperty("java.version"))
+    )
+
+  @internal
+  def bspJvmBuildTargetTask: Task[JvmBuildTarget] = Task.Anon {
```

Comment:
This new impl will miss any customizations. It should instead reuse the value of deprecated method.